packages/uni_lint/lib/uni_lint.dart

```diff
@@ -0,0 +1,59 @@
+import 'package:analyzer/error/listener.dart';
+import 'package:custom_lint_builder/custom_lint_builder.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+
+PluginBase createPlugin() => _ExampleLinter();
+
+class _ExampleLinter extends PluginBase {
+  @override
+  List<LintRule> getLintRules(CustomLintConfigs configs) => [
+        StringLiteralsLint(),
+      ];
+}
+
+class StringLiteralsLint extends DartLintRule {
+  StringLiteralsLint() : super(code: _code);
+
+  static const _code = LintCode(
+    name: 'string_literals_lint',
+    problemMessage: 'There is a string literal inside a widget',
+  );
+
+  @override
+  void run(
+    CustomLintResolver resolver,
+    ErrorReporter reporter,
+    CustomLintContext context,
+  ) {
+    context.registry.addVariableDeclaration((node) {
```

Comment:
The fact that this is adding a method for `variableDeclaration` makes me wonder:

```dart
final myString = "This will trigger a linting warning, right?"
```

```dart
return Container(
  child: Text("But this will not trigger a linting warning, right?")
)
```


In my opinion, the way this should work is go through every string literal (something like `context.registry.addStringLiteral` - disclaimer, I don't know if this exists!) and check if it within a widget class.

Reply:
To speed up the linting, we can instead target `ClassDeclaration`s, with something like `context.registry.addClassDeclaration` (disclaimer, I don't know if this exists!) and then, for each class declaration, get all descendant string literals using something like `node.descendantsOfType<StringLitera>()` (disclaimer, I don't know if this exists!) and then create a linting warning at every string literal.